openupgrade_scripts/scripts/account/14.0.1.1/pre-migration.py

```diff
@@ -0,0 +1,379 @@
+                "payment_reference",
+            ),
+            (
+                "account.move",
+                "account_move",
+                "invoice_sent",
+                "is_move_sent",
+            ),
+            (
+                "account.move",
+                "account_move",
+                "type",
+                "move_type",
+            ),
+            (
+                "account.move",
+                "account_move",
+                "invoice_payment_state",
+                "payment_state",
+            ),
+            (
+                "account.move.line",
+                "account_move_line",
+                "tag_ids",
+                "tax_tag_ids",
+            ),
+            (
+                "res.company",
+                "res_company",
+                "account_onboarding_sample_invoice_state",
+                "account_onboarding_create_invoice_state",
+            ),
+            (
+                "res.company",
+                "res_company",
+                "accrual_default_journal_id",
+                "automatic_entry_default_journal_id",
+            ),
+        ],
+    )
+
+
+def m2m_tables_account_journal_renamed(env):
+    openupgrade.rename_tables(
+        env.cr,
+        [
+            ("account_account_type_rel", "journal_account_control_rel"),
+            ("account_journal_type_rel", "journal_account_type_control_rel"),
+        ],
+    )
+
+
+def remove_constrains_reconcile_models(env):
+    openupgrade.remove_tables_fks(
+        env.cr,
+        [
+            "account_reconcile_model_analytic_tag_rel",
+            "account_reconcile_model_account_tax_rel",
+            "account_reconcile_model_template_account_tax_template_rel",
+            "account_reconcile_model_tmpl_account_tax_bis_rel",
+        ],
+    )
+    openupgrade.logged_query(
+        env.cr,
+        """
+        ALTER TABLE account_reconcile_model_analytic_tag_rel
+        ALTER COLUMN account_reconcile_model_id DROP NOT NULL
+        """,
+    )
+    openupgrade.logged_query(
+        env.cr,
+        """
+        ALTER TABLE account_reconcile_model_analytic_tag_rel
+        ADD COLUMN account_reconcile_model_line_id integer
+        """,
+    )
+
+
+def copy_fields(env):
+    openupgrade.rename_columns(
+        env.cr,
+        {
+            "account_journal": [
+                ("default_credit_account_id", None),
+                ("default_debit_account_id", None),
+            ],
+        },
+    )
+    if openupgrade.column_exists(env.cr, "account_move", "move_type"):
+        # see account_tax_balance of OCA
+        openupgrade.rename_columns(
+            env.cr,
+            {
+                "account_move": [
+                    ("move_type", None),
+                ],
+            },
+        )
+
+
+def add_move_id_field_account_bank_statement_line(env):
+    if not openupgrade.column_exists(
+        env.cr, "account_bank_statement_line", "currency_id"
+    ):
+        openupgrade.logged_query(
```

Comment:
getting an error "column statement_line exists in relation 'account_move', checking if it's data-specific or a recurring problem. will follow up in 30m